contracts/LiquidityBridgeContract.sol

```diff
@@ -225,69 +279,134 @@ contract LiquidityBridgeContract {
+		// SDL: Because a bug in the bridge, only the Low significant 4 bytes
+		// in height are considered. To protect from unintended vulns,
+		// we should require that height < 2^32.
+        require(height < uint256(MAX_INT32), "Height must be lower than 2^31");
+		
+		// SDL: rename transferredAmountOrErrorCode to transferredAmountOrErrorCodeOrErrorCode
+		// Too many future checks depends on this variable being an error code
+		// Do not hide this fact.
+        int256 transferredAmountOrErrorCode = registerBridge(quote, btcRawTransaction, partialMerkleTree, height, quoteHash);
+
+		// Return codes should be stored in consts. 
+        require(transferredAmountOrErrorCode != BRIDGE_UNPROCESSABLE_TX_VALIDATIONS_ERROR, "Failed to validate BTC transaction");
+        require(transferredAmountOrErrorCode != BRIDGE_UNPROCESSABLE_TX_ALREADY_PROCESSED_ERROR_CODE, "Transaction already processed");
+        require(transferredAmountOrErrorCode != BRIDGE_UNPROCESSABLE_TX_VALUE_ZERO_ERROR, "Transaction value is zero");
+        require(transferredAmountOrErrorCode != BRIDGE_GENERIC_ERROR, "Bridge error");
+
+		
+        if (shouldPenalizeLP(quote, transferredAmountOrErrorCode, callRegistry[quoteHash].timestamp, height)) {
             collateral[quote.liquidityProviderRskAddress] -= quote.penaltyFee;
             emit Penalized(quote.liquidityProviderRskAddress, quote.penaltyFee, quoteHash);
             
             // pay reward to sender
-            uint256 reward = quote.penaltyFee / rewardR;    
-            increaseBalance(msg.sender, reward);        
+			// SDL: rewardP is a divisor... why not making more granual, such as 
+			// multiplying penaltyFee by 10 before dividing ?
+			// SDL: rename reward punisherReward or something alike.
+            uint256 punisherReward = quote.penaltyFee * rewardP / 100;    
+            increaseBalance(msg.sender, punisherReward);        
             
             // burn the rest of the penalty
-            (bool success, ) = payable(0x00).call{value : quote.penaltyFee - reward}("");
-            require(success, "Could not burn penalty");            
+			// SDL: Why it is necessary to send it to another contract 0x00
+			// for burning. Leaving it in this contract is enough.       
         }
-
-        if (transferredAmount == -200 || transferredAmount == -100) {
+		// SDL: -100 and -200 should be final constants defined with an
+		// associated identifier. What's the difference between them?
+		
+		
+        if (transferredAmountOrErrorCode == BRIDGE_REFUNDED_LP_ERROR_CODE || transferredAmountOrErrorCode == BRIDGE_REFUNDED_USER_ERROR_CODE) {
             // Bridge cap exceeded
-            callRegistry[quoteHash].timestamp = 0;
-            callRegistry[quoteHash].success = false;
-            emit BridgeCapExceeded(quoteHash, transferredAmount);
-            return transferredAmount;
+			// There is a Solidity sentence called "delete" that I think
```

Comment:
Remember to check if delete is what you need here

Reply:
Since I added a check to avoid processing quotes twice this is no longer needed unless there is an advantage in freeing memory.